internal/driver/driver.go

```diff
@@ -606,23 +607,37 @@ func (d *Driver) newDevice(name string, protocols map[string]models.ProtocolProp
 		return nil, errors.NewCommonEdgeX(errors.KindServerError,
 			fmt.Sprintf("could not find the serial number of the device on the specified path: %s", fdPath), err)
 	}
-	psn := protocols[UsbProtocol][SerialNumber].(string)
-	pcn := protocols[UsbProtocol][CardName].(string)
-	// pre-defined devices may not include serial number information
-	if len(psn) == 0 {
+
+	// if the user provided a serial number or card name, but it does not match the device's serial number or card name,
+	// then return an error, as this may not be the correct device
+	psn, psnOK := protocols[UsbProtocol][SerialNumber].(string)
```

Comment:
Should also validate `protocols[UsbProtocol]` exists before index on it.

Reply:
That should already be handled by the validateDevice call I believe.

Reply:
Shouldn't validateDevice also ensure SerialNumber and CardName are present?

Reply:
`SerialNumber` and `CardName` are optional parameters.